runtime/lua/vim/lsp/_watchfiles.lua

```diff
@@ -264,15 +264,27 @@ function M.register(reg, ctx)
     end
   end
 
-  local watching = {}
-  for _, w in ipairs(watch_regs) do
-    if not watching[w.base_dir] then
-      watching[w.base_dir] = true
-      table.insert(
-        cancels[client_id][reg.id],
-        M._watchfunc(w.base_dir, { uvflags = { recursive = true } }, callback(w.base_dir))
-      )
+  for base_dir, watches in pairs(watch_regs) do
+    local include_patterns = {}
+    for _, w in ipairs(watches) do
```

Comment:
maybe there can also use `vim.iter` syntax suger make it simply.

Reply:
@glepnir I took a look through `vim.iter` but couldn't come up with anything that seemed to make this much simpler. Did you have an idea of what this could possibly look like?

Reply:
if backport to 0.9 .ignore this :)